auth.business/src/main/java/com/github/pplociennik/auth/business/mailing/EmailService.java

```diff
@@ -0,0 +1,41 @@
+package com.github.pplociennik.auth.business.mailing;
+
+import com.github.pplociennik.auth.business.mailing.domain.EmailConfirmationSendingDO;
+import com.github.pplociennik.auth.business.mailing.message.ConfirmationRequestStrategy;
+import com.github.pplociennik.auth.mailing.MailingService;
+import com.github.pplociennik.auth.mailing.ports.MessageContentProvider;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.lang.NonNull;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * A service sharing methods for project specific mailing functionalities.
+ *
+ * @author Created by: Pplociennik at 07.05.2022 22:24
+ */
+class EmailService {
+
+    private final MessageContentProvider contentProvider;
+    private final MailingService mailingService;
+
+    @Autowired
+    EmailService( @NonNull MessageContentProvider aContentProvider, MailingService aMailingService ) {
+        mailingService = requireNonNull( aMailingService );
+        contentProvider = requireNonNull( aContentProvider );
+    }
+
+    /**
+     * Prepares an email confirmation request, which contains a confirmation link.
+     *
+     * @param aSendingDO
+     *         a data object providing necessary information.
+     */
+    void prepareAndSendEmailConfirmationRequest( @NonNull EmailConfirmationSendingDO aSendingDO ) {
+        var creator = new ConfirmationRequestStrategy( contentProvider );
```

Comment:
Nie może to tak zostać! Do refactoringu! Przekazywana jest implementacja portu z infrastructure do zwykłej POJO, która jest tylko strategią tworzenia wiadomości!

Reply:
Powoduje to dodatkowe problemy, jak to, że metoda w implementacji (notabene package-private) musi być publiczna!

Reply:
done

Reply:
ale do przejrzenia... Nie wiem czy to wszystko nie jest nieco przekombinowane